src/test.rs

```diff
@@ -394,6 +395,70 @@ fn layer2() {
 }
 
 #[test]
+fn layer2_timeouts() {
+    use std::time::Duration;
+    use std::io::ErrorKind;
+    use datalink;
+    use datalink::Channel::Ethernet;
+    use packet::ethernet::{EtherTypes, MutableEthernetPacket};
+
+    const ETHERNET_HEADER_LEN: usize = 14;
+
+    let interface = get_test_interface();
+
+    let mut packet = [0u8; ETHERNET_HEADER_LEN + IPV4_HEADER_LEN + UDP_HEADER_LEN + TEST_DATA_LEN];
+
+    {
+        let mut ethernet_header = MutableEthernetPacket::new(&mut packet[..]).unwrap();
+        ethernet_header.set_source(interface.mac_address());
+        ethernet_header.set_destination(interface.mac_address());
+        ethernet_header.set_ethertype(EtherTypes::Ipv4);
+    }
+
+    build_udp4_packet(&mut packet[..],
+                      ETHERNET_HEADER_LEN as usize,
+                      "l2tt",
+                      Some(&interface));
+
+    let (tx, rx) = channel();
+
+    let cfg = datalink::Config {
+        write_buffer_size: 4096,
+        read_buffer_size: 4096,
+        channel_type: datalink::ChannelType::Layer2,
+        bpf_fd_attempts: 100,
+        read_timeout: Some(Duration::from_millis(300)),
+        write_timeout: Some(Duration::from_millis(100))
+    };
```

Comment:
Rather than filling in all these fields manually, you can use [record update syntax](https://doc.rust-lang.org/book/structs.html#update-syntax), eg:
```rust
SomeStruct {
    field1: value,
    ..Default::default()
}
```
This saves filling out all the irrelevant defaults, and makes it clear which fields you care about.